src/hotspot/share/gc/z/zNMethod.cpp

```diff
@@ -236,28 +236,40 @@ void ZNMethod::nmethod_oops_do(nmethod* nm, OopClosure* cl) {
 
 class ZNMethodToOopsDoClosure : public NMethodClosure {
 private:
-  OopClosure* const _cl;
-  const bool        _should_disarm_nmethods;
+  OopClosure* const        _cl;
+  const ZNMethodEntry      _entry;
+  BarrierSetNMethod* const _bs_nm;
 
 public:
-  ZNMethodToOopsDoClosure(OopClosure* cl, bool should_disarm_nmethods) :
+  ZNMethodToOopsDoClosure(OopClosure* cl, ZNMethodEntry entry) :
       _cl(cl),
-      _should_disarm_nmethods(should_disarm_nmethods) {}
+      _entry(entry),
+      _bs_nm(BarrierSet::barrier_set()->barrier_set_nmethod()) {}
 
   virtual void do_nmethod(nmethod* nm) {
+    if (_entry == ZNMethodEntry::PreBarrier) {
+      // Apply entry barrier before proceeding with closure
+      _bs_nm->nmethod_entry_barrier(nm);
+    }
+
     ZLocker<ZReentrantLock> locker(ZNMethod::lock_for_nmethod(nm));
     if (!nm->is_alive()) {
       return;
     }
 
-    if (_should_disarm_nmethods) {
+    if (_entry == ZNMethodEntry::VerifyDisarmed) {
+      // Only verify
+      assert(!ZNMethod::is_armed(nm), "Must be disarmed");
+
```

Comment:
Extra white space can be removed.

Reply:
If you insist. I think this style makes the code easier to read.

Reply:
Yes please, to me it looks unintended, especially since we don't use that style elsewhere. Maybe flipping the `if (VerifyDisarmed)` and `else if(Disarm)` would make it read better?